psyneulink/core/scheduling/condition.py

```diff
@@ -152,3 +155,129 @@ def _dict_summary(self):
         )
     ]
 }
+
+
+class Threshold(graph_scheduler.condition._DependencyValidation, Condition):
+    """Threshold
+
+    Attributes:
+
+        dependency
+            the node on which the Condition depends
+
+        parameter
+            the name of the parameter of **dependency** whose value is
+            to be compared to **threshold**
+
+        threshold
+            the fixed value compared to the value of the **parameter**
+
+        comparator
+            the string comparison operator determining the direction or
+            type of comparison of the value of the **parameter**
+            relative to **threshold**
+
+        indices
+            if specified, a series of indices that reach the desired
+            number given an iterable value for **parameter**
+
+        atol
+            absolute tolerance for the comparison
+
+        rtol
+            relative tolerance (to **threshold**) for the comparison
+
+
+    Satisfied when:
+
+        The comparison between the value of the **parameter** and
+        **threshold** using **comparator** is true
+
+    Notes:
+
+        The comparison must be done with single numbers. If the value of
+        **parameter** contains more than one item, **indices** must be
+        specified.
+    """
+
+    def __init__(
+        self,
+        dependency,
+        parameter,
+        threshold,
+        comparator,
+        indices=None,
+        atol=0,
+        rtol=0,
+    ):
+        if comparator not in comparison_operators:
+            raise graph_scheduler.ConditionError(
+                f'Operator must be one of {list(comparison_operators.keys())}'
+            )
+
+        if parameter not in dependency.parameters:
+            raise graph_scheduler.ConditionError(
+                f'{dependency} has no {parameter} parameter'
+            )
+
+        if (
+            indices is not None
+            and not isinstance(indices, graph_scheduler.TimeScale)
+            and not isinstance(indices, collections.abc.Iterable)
+        ):
+            indices = [indices]
+
+        tolerance_value = rtol * abs(threshold) + atol
```

Comment:
noticed one more nit.
You're calculating tolerance value here, but at the same time passing both atol and rtol to 'func'